ApplicationInsights/JavaScriptSDK/ApplicationInsights.ts

```diff
@@ -13,24 +13,23 @@ import { IPlugin, IConfiguration, IAppInsightsCore, ITelemetryPlugin, CoreUtils,
 import { TelemetryContext } from "./TelemetryContext";
 import { PageVisitTimeManager } from "./Telemetry/PageVisitTimeManager";
 import { IAppInsights } from "../JavascriptSDK.Interfaces/IAppInsights";
-import { IPageViewTelemetry } from "../JavascriptSDK.Interfaces/IPageViewTelemetry";
+import { IPageViewTelemetry, IPageViewTelemetryInternal } from "../JavascriptSDK.Interfaces/IPageViewTelemetry";
 import { ITelemetryConfig } from "../JavaScriptSDK.Interfaces/ITelemetryConfig";
 import { TelemetryItemCreator } from "./TelemetryItemCreator";
 
 "use strict";
 
 export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IAppInsightsInternal {
 
-    public static defaultIdentifier = "ApplicationInsightsAnalytics";
-    public processTelemetry: (env: ITelemetryItem) => void;
+    public static defaultIdentifier = "ApplicationInsightsAnalytics";    
     public identifier: string;
-    public setNextPlugin: (next: ITelemetryPlugin) => void;
     priority: number;
     public initialize: (config: IConfiguration, core: IAppInsightsCore, extensions: IPlugin[]) => void;
```

Comment:
This one, will still throw errors for classes extending, when calling super.initialize()

Reply:
Yup, Abhi wants to discuss this. Both approaches of implementing the required interface are valid. But obviously one limits extensibility for apis.